overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,96 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// StoreConfigSnapshotMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+// The caller is responsible for locking the state.
+func StoreConfigSnapshotMaybe(st *state.State, snapName string, rev snap.Revision) error {
```

Comment:
Yes, I like it. How about the *Maybe in the names though? I'd keep them to highlight the fact that these methods are tolerent to missing data.